src/omit.ts

```diff
@@ -12,10 +13,10 @@ import { purry } from './purry';
  * @dataFirst
  * @category Object
  */
-export function omit<T extends object, K extends keyof T>(
+export function omit<T extends Record<PropertyKey, unknown>, K extends keyof T>(
   data: T,
   propNames: ReadonlyArray<K>
-): Omit<T, K>;
+): Unwrap<Omit<T, K>>;
```

Comment:
Was this type broken? I thought we are only encountering issues with the dataLast implementation?

Reply:
I just watched the `object` type and auto updated it 😅